test/wokwi/prepare-firmware.sh

```
@@ -0,0 +1,218 @@
+    echo "4. For ESP-IDF framework, bootloader should be auto-generated"
+    echo ""
+    echo "If files are in unexpected locations, the script will show them above."
+    exit 1
+fi
+
+echo "✅ All required files found!"
+echo ""
+
+echo "Creating combined firmware image for Wokwi..."
+echo "Using files from $ENV_NAME build:"
+echo "  - Bootloader: $(basename $(dirname "$BOOTLOADER_BIN"))/$(basename "$BOOTLOADER_BIN")"
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)